### ISSUE TITLE
Short Address assign

### ISSUE THREAD

**Issue by rofacz** [Jon-Bright/dalifi#4]
Hello,
is there any reason why the devices are not assigned by consecutive short addresses?
First device gets address 0, next 2,....
This is caused in  daliAddr* Dali::reAddressLamps(byte *num)
by:
 *num = shortAddr;
  for (byte b = 0; b < shortAddr; b++) {
    ret[b] = b << 1;
  }
DALI accepts 64 devices on the bus (0..63), but there is the out of range formula.
If I am correct, the ret[b] = b; shoud be used.

**Comment by Jon-Bright** [comment]
Hi,

tl;dr: all 64 addresses are usable, the short addresses getting assigned are correct, just the byte being returned is shifted up by one bit, for reasons.

The address byte for DALI commands with short addresses is:

* Bit 7 (top bit): 0
* Bits 6-1: the short address (6 bits, 64 addresses)
* Bit 0 (low bit): 1 for standard commands, 0 for DAPC (direct arc power control) commands

Sending of the address to the lamp is happening in `findDevice`: https://github.com/Jon-Bright/dalifi/blob/30ad1835ea71426921f120f218fd47839ef958a7/library/dali.cpp#L572

The DALI "set short address" command also takes the address shifted up by one bit.

By storing the address pre-shifted, we avoid having to shift it everywhere else, when sending commands.

In `sendLampOff`, for example, you can see that we're setting bit 0 to 1, because this is a standard command: https://github.com/Jon-Bright/dalifi/blob/30ad1835ea71426921f120f218fd47839ef958a7/library/dali.cpp#L424
By contrast in `sendDapc`, nothing is getting set in the bottom bit, meaning it stays as 0 and this is a DAPC command: https://github.com/Jon-Bright/dalifi/blob/30ad1835ea71426921f120f218fd47839ef958a7/library/dali.cpp#L438

The reason DAPC commands are different: they're sent to a specific lamp (or group of lamps, etc.), but they need to send a level. Thus, the command itself is encoded in the bottom bit of the address byte, while the level is encoded in the opcode byte.

Hope this helps!

Thanks

Jon

**Comment by rofacz** [comment]
Thans for detailed description!
Meanwhile I changed little bit the addressing in library to get better compatibility with my app.
